values.go

```diff
@@ -1974,3 +1974,63 @@ func (v Function) String() string {
 	// TODO: include function type
 	return format.Function("(...)")
 }
+
+// ValueWithCachedTypeID recursively caches type ID of value v's type.
+// This is needed because each type ID is lazily cached on
+// its first use in ID() to avoid performance penalty.
+func ValueWithCachedTypeID[T Value](value T) T {
```

Comment:
_Note: This method is only used for tests, but is added here to avoid circular dependencies._